### PR TITLE
Improve messages on adding hosts errors

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

This PR improves the error messages displayed when adding a host fails

Fixes: #5677

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### Screenshots (if appropriate):
Screen Shot 2022-02-15 at 10 58 57
Screen Shot 2022-02-15 at 10 59 33
Screen Shot 2022-02-15 at 11 01 24

### How Has This Been Tested?
Tested on KVM env:
- Add host in different scenarios: unreachable IP address, invalid credentials, valid credentials but not KVM enabled

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

File: server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java [hunk 1]
```diff
@@ -372,13 +375,12 @@ private void setupAgentSecurity(final Connection sshConnection, final String age
             throw e;
         } catch (Exception e) {
             String msg = " can't setup agent, due to " + e.toString() + " - " + e.getMessage();
-            s_logger.warn(msg);
+            s_logger.warn(msg, e);
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java:378]:
```suggestion
            s_logger.warn(msg);
            if (s_logger.isDebugEnabled()) {
                s_logger.debug(msg, e);
            }
```

File: server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java [hunk 2]
```diff
@@ -273,8 +273,11 @@ private void setupAgentSecurity(final Connection sshConnection, final String age
             }
 
             if (!SSHCmdHelper.sshExecuteCmd(sshConnection, "ls /dev/kvm")) {
-                s_logger.debug("It's not a KVM enabled machine");
-                return null;
+                String errorMsg = "It's not a KVM enabled machine";
```
Comment (GutoVeronezi) [inline server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java:276]:
```suggestion
                String errorMsg = "This machine does not have KVM enabled.";
```

File: server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java [hunk 3]
```diff
@@ -335,9 +338,9 @@ private void setupAgentSecurity(final Connection sshConnection, final String age
                 setupAgentCommand = "sudo cloudstack-setup-agent ";
             }
             if (!SSHCmdHelper.sshExecuteCmd(sshConnection, setupAgentCommand + parameters)) {
-                s_logger.info("cloudstack agent setup command failed: "
-                        + setupAgentCommand + parameters);
-                return null;
+                String errorMsg = "cloudstack agent setup command failed: " + setupAgentCommand + parameters;
```
Comment (GutoVeronezi) [inline server/src/main/java/com/cloud/hypervisor/kvm/discoverer/LibvirtServerDiscoverer.java:341]:
```suggestion
                String errorMsg = String.format("CloudStack Agent setup through command [%s] with parameters [%s] failed.", setupAgentCommand, parameters);
```

File: server/src/main/java/com/cloud/resource/ResourceManagerImpl.java [hunk 4]
```diff
@@ -807,9 +807,16 @@ public Discoverer getMatchingDiscover(final Hypervisor.HypervisorType hypervisor
             try {
                 resources = discoverer.find(dcId, podId, clusterId, uri, username, password, hostTags);
             } catch (final DiscoveryException e) {
-                throw e;
+                String errorMsg = "Unable to add the host: " + e.getMessage();
```
Comment (GutoVeronezi) [inline server/src/main/java/com/cloud/resource/ResourceManagerImpl.java:810]:
```suggestion
                String errorMsg = String.format("Could not add host at [%s] with zone [%s], pod [%s] and cluster [%s] due to [%s].", uri, dcId, podId, clusterId, e.getMessage());
```

**Comment by nvazquez** [comment]
Thanks for reviewing @DaanHoogland @GutoVeronezi - suggestions applied
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

File: server/src/main/java/com/cloud/resource/ResourceManagerImpl.java [hunk 5]
```diff
@@ -863,8 +872,9 @@ public Discoverer getMatchingDiscover(final Hypervisor.HypervisorType hypervisor
             s_logger.warn(msg);
             throw new DiscoveryException(msg);
         }
-        s_logger.warn("Unable to find the server resources at " + url);
-        throw new DiscoveryException("Unable to add the host");
+        String errorMsg = "Unable to find the server resources at " + url;
+        s_logger.warn(errorMsg);
+        throw new DiscoveryException("Unable to add the host: " + errorMsg);
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/resource/ResourceManagerImpl.java:877]: double "Unable" ends up in this message!?! (double negative...)

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan test

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Comment by nvazquez** [comment]
Ping for review @davidjumani @Pearl1594

**Review by Pearl1594 — APPROVED** [review]
Changes LGTM.

**Pull request merged**